axelor-base/src/main/java/com/axelor/apps/base/service/ProductPackagingServiceImpl.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2005-2025 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package com.axelor.apps.base.service;
+
+import com.axelor.apps.base.AxelorException;
+import com.axelor.apps.base.db.Product;
+import com.axelor.apps.base.db.ProductPackaging;
+import com.axelor.apps.base.db.Unit;
+import com.axelor.apps.base.db.repo.UnitRepository;
+import com.axelor.apps.base.service.app.AppBaseService;
+import com.google.inject.Inject;
+import java.math.BigDecimal;
+
+public class ProductPackagingServiceImpl implements ProductPackagingService {
+
+  protected static final String UNIT_KG = "kg";
+
+  protected final UnitConversionService unitConversionService;
+  protected final UnitRepository unitRepository;
+
+  @Inject
+  public ProductPackagingServiceImpl(
+      UnitConversionService unitConversionService, UnitRepository unitRepository) {
+    this.unitConversionService = unitConversionService;
+    this.unitRepository = unitRepository;
+  }
+
+  @Override
+  public BigDecimal computePackagingMaxTotalWeight(ProductPackaging productPackaging)
+      throws AxelorException {
+    BigDecimal packagingWeight = getConvertedGrossMass(productPackaging.getPackaging());
+    BigDecimal productTotalWeight = getProductTotalWeight(productPackaging);
+    return packagingWeight.add(productTotalWeight);
+  }
+
+  @Override
+  public boolean validateTotalWeight(ProductPackaging productPackaging) throws AxelorException {
+    Product packaging = productPackaging.getPackaging();
+    if (packaging == null) {
+      return true;
+    }
+    BigDecimal productTotalWeight = getProductTotalWeight(productPackaging);
+    if (productTotalWeight.compareTo(packaging.getMaxWeight()) > 0) {
+      return false;
+    }
+    return true;
```

Comment:
By changing the method name by isTotalWeightNotValid,
You can just return productTotalWeight.compareTo(packaging.getMaxWeight()) > 0
and it will be more clear.